feat(YellowNote): add source 小黄书

marioplus opened the pull request:
Checklist:

- [x] Updated `extVersionCode` value in `build.gradle` for individual extensions
- [x] Updated `overrideVersionCode` or `baseVersionCode` as needed for all multisrc extensions
- [ ] Referenced all related issues in the PR body (e.g. "Closes #xyz")
- [x] Added the `isNsfw = true` flag in `build.gradle` when appropriate
- [ ] Have not changed source names
- [ ] Have explicitly kept the `id` if a source's name or language were changed
- [x] Have tested the modifications by compiling and running the extension through Android Studio
- [x] Have removed `web_hi_res_512.png` when adding a new extension

marioplus committed:
feat(YellowNote): add source 小黄书

marioplus committed:
fix(YellowNote): reverse chapter sorting order

AwkwardPeak7 commented on src/all/yellownote/src/eu/kanade/tachiyomi/extension/all/yellownote/YellowNote.kt:
```diff
@@ -0,0 +1,127 @@
+        val categorySelector = filters.firstInstance<YellowNoteFilters.CategorySelector>()
+        val uriPart = when {
+            query.isBlank() -> "$baseUrl${categorySelector.toUrlPart()}"
+            else -> "$baseUrl/photos/keyword-$query"
```
use `HttpUrl.Builder` to properly encode the query

marioplus replied:
the query parameter is located in the path. should I use `GET($urlPart.toHttpURL())`?

AwkwardPeak7 replied:
using `addPathSegment("keyword-$query")` from `HttpUrl.Builder` should be enough

AwkwardPeak7 commented on src/all/yellownote/src/eu/kanade/tachiyomi/extension/all/yellownote/YellowNotePreferences.kt:
```diff
@@ -0,0 +1,42 @@
+
+    private const val DEFAULT_BASE_URL = "https://xchina.co"
+
+    internal fun SharedPreferences.baseUrl(subdomain: String?): String {
```
is a baseurl pref needed?
also are the mirrors known? if so make a mirror pref, without option to enter custom url

marioplus replied:
the mirrors is unknown. due to firewall issues in China, the website may change its domain name after a period of time

AwkwardPeak7 replied:
then please implement mirror pref in [such a way](https://github.com/keiyoushi/extensions-source/blob/71d5f45eacbf5e41db821f08af1df638ecd8c5a6/src/ar/teamx/src/eu/kanade/tachiyomi/extension/ar/teamx/TeamX.kt#L230-L261)

basically save current mirror from code in another pref value, and compare that to one in code, if they are different (ext updated to change it), change both in pref to match it

marioplus committed:
feat(YellowNote): improve domain configuration and URL handling

- Implement dual domain configuration (default + override)
  - Default settings refresh when code updates occur
  - Override settings maintain custom configurations
- Enhance URL processing using HttpUrl
  - Fix encoding issues in query URLs
  - Improve URL parsing reliability

marioplus committed:
feat(YellowNote): adapter /amateurs.html
